tracer/src/Datadog.Trace.Tools.Runner/Checks/Linux/ProcessEnvironmentLinux.cs

```diff
@@ -0,0 +1,35 @@
+// <copyright file="ProcessEnvironmentLinux.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+
+namespace Datadog.Trace.Tools.Runner.Checks.Linux
+{
+    internal class ProcessEnvironmentLinux
+    {
+        public static IReadOnlyDictionary<string, string> ReadVariables(Process process)
+        {
+            /*
+                   /proc/[pid]/environ
+                          This file contains the environment for the process. The entries are separated by
+                          null bytes ('\0'), and there may be a null byte at the end.
+            */
+
+            var path = $"/proc/{process.Id}/environ";
+
+            var result = new Dictionary<string, string>();
+
+            foreach (var line in File.ReadAllText(path).Split('\0'))
```

Comment:
nit: maybe remove the empty ones, e.g. from final null byte?
```suggestion
            foreach (var line in File.ReadAllText(path).Split('\0', StringSplitOptions.RemoveEmptyEntries))
```

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/2323/commits/786cea46f436ee5d91b283db1f93ca30207707aa